docs/gke-autopilot.md

```diff
@@ -0,0 +1,268 @@
+
+- **Autopilot mode**, which is designed to reduce the operational cost of
+managing clusters, apply best security practices to harden your clusters, and
+optimize them for production to yield higher workload availability.
+
+- **Standard mode** in which you are responsible for provisioning, managing, and
+monitoring the nodes of your physical cluster.
+
+In this tutorial, we'll demonstrate how to deploy the Online Boutique
+microservices demo using GKE Autopilot mode. You'll see firsthand how using
+Autopilot mode is simple and straightforward.
+
+In the final section, we'll explain the benefits of using GKE Autopilot in more
+detail.
+
+## Getting Started
+
+Getting started is easy and we'll show two ways to create your GKE Autopilot
+cluster.
+
+- Using Cloud Console.
+- Using the command line.
+
+Once you have a created a cluster, you can go to the section on [Deploying the
+demo].
+
+> Whether you use Cloud Console or the command line to create the cluster, you
+will still need to use `kubectl` at the command line to deploy the application.
+>
+> The instructions assume you already have [Google Cloud SDK (gcloud)] and
+[kubectl] (which can be optionally installed as a `gcloud` component).
+>
+> If you prefer, you can use [Cloud Shell] for a hosted terminal environment
+that already has these tools installed for you.
+
+
+### Select or create a project using Cloud Console
+
+1. In the Google Cloud Console, go to the [project selector] page. Either select
+an existing project or create a new one.
+
+1. Next, using the left navigation menu, choose **Kubernetes Engine**.
+
+1. If prompted, click to **ENABLE** the Kubernetes API.
+
+1. On the **Kubernetes Engine Clusters** page, click the **CREATE** button.
+
+1. The **Create Cluster** dialog will prompt you to select a cluster mode. Click
+the **CONFIGURE** button.
+
+1. You will be prompted to **Create an Autopilot cluster**. You can accept all
+the defaults.
+
+1. Once the cluster is created, click on the name link to go to the cluster
+**DETAILS** tab. Click the **CONNECT** button to get the command you'll use to
+connect to the cluster using the command line. Copy the command and paste it
+to your command line and press enter.
+
+Now you're ready to deploy the demo.
+
+### Select or create a project using the command line
+
+The following steps are purely administrative. You need to log in and identify
+the project you want to work with.
+
+```text
+gcloud auth login
+```
+
+Then either choose an existing project or create a new one.
+
+#### Select an existing project
+
+```text
+PROJECT_ID=my-project-id
+gcloud services enable container.googleapis.com --project $PROJECT_ID
+```
+
+This assumes you have billing set for the project. If not, see the next section
+and look for the `gcloud beta billing` command to link billing up.
+
+#### Create a new project
+
+> You will need to know your [billing ID].
+
+```text
+PROJECT_NAME=my-project-name
+gcloud projects create $PROJECT_NAME
```

Comment:
When I ran: 
```
gcloud projects create $PROJECT_NAME
```
I got the following error:
```
ERROR: (gcloud.projects.create) Operation [cp.5078032222408546781] failed: 7: Permission 'resourcemanager.projects.create' denied on parent resource 'organizations/XXXXXXXXXXXXX'
```
I got this error because `gcloud` was using the `google.com` organization by default.
So I had to explicitly specify my organization. This worked for me:
```
gcloud projects create $PROJECT_NAME --organization=XXXXXXXXXXXX
```

**Suggestion (Optional)**

This isn't absolutely necessary (because I am not sure if our users will be juggling multiple organization), but it might be worth adding:
* something like `--organization=XXXXXXXXXXXX` to the command and 
* a note containing a about needing your organization ID, with a link to [this doc](https://cloud.google.com/resource-manager/docs/creating-managing-organization#retrieving_your_organization_id)
 (similar to the "You will need ... billing ID" note).